yt_dlp/extractor/reddit.py

```diff
@@ -306,7 +349,8 @@ def add_thumbnail(src):
                 'video_id', default=display_id)
 
             dash_playlist_url = playlist_urls[0] or f'https://v.redd.it/{video_id}/DASHPlaylist.mpd'
-            hls_playlist_url = playlist_urls[1] or f'https://v.redd.it/{video_id}/HLSPlaylist.m3u8'
+            hls_playlist_url = re.sub(r'(f=[^&]+)', r'\1%2CsubsAll', playlist_urls[1]) or f'https://v.redd.it/{video_id}/HLSPlaylist.m3u8?f=hd%2CsubsAll'
+            caption_url = f'https://v.redd.it/{video_id}/wh_ben_en.vtt'
```

Comment:
use `parse_qs` and `update_url_query`, and move the `caption_url` assignment down below where we use it
```suggestion
            hls_playlist_url = playlist_urls[1] or f'https://v.redd.it/{video_id}/HLSPlaylist.m3u8'
            qs = traverse_obj(parse_qs(url), {'f': ('f', 0, {lambda x: ','.join([x, 'subsAll']) if x else 'hd,subsAll'})})
            hls_playlist_url = update_url_query(hls_playlist_url, qs)
```